galsim/chromatic.py

```diff
@@ -1359,3 +1376,286 @@ def withScaledFlux(self, flux_ratio):
+            self.stepK_vals = [ obj.stepK() for obj in objs ]
+            self.maxK_vals = [ obj.maxK() for obj in objs ]
+
+            # Finally, now that we have an image scale and size, draw all the images.  Note that
+            # `no_pixel` is used (we want the object on its own, without a pixel response).
+            self.ims = [ obj.drawImage(scale=dx, nx=n_im, ny=n_im, method='no_pixel')
+                         for obj in objs ]
+
+    def withSED(self, sed='flat'):
+        """
+        Make a new InterpolatedChromaticObject with the requested SED.
+
+        @param sed    Either a galsim.SED object to assign to this object as its SED, or a string
+                      'flat' which means a flat SED (equal weight to all wavelengths).  'flat' is
+                      what should be used for PSFs, so they do not modify the flux of the objects
+                      with which they are being convolved. 
+                      [default: 'flat']
+        @return a new InterpolatedChromaticObject
+        """
+        new = self.copy()
+        if sed == 'flat':
+            new.SED = lambda w: 1.0
+        else:
+            new.SED = sed
+        return new
+
+    def evaluateAtWavelength(self, wave, force_eval=False):
+        """
+        Evaluate this InterpolatedChromaticObject at a particular wavelength, either using
+        interpolation or via direct calculation, depending on how the object was originally
+        instantiated.
+
+        @param wave     Wavelength in nanometers.
+
+        @returns the monochromatic object at the given wavelength, as an InterpolatedImage.
+        """
+        if self.waves is not None and not force_eval:
+            im, stepk, maxk = self._imageAtWavelength(wave)
+            return galsim.InterpolatedImage(im, _force_stepk=stepk, _force_maxk=maxk)
+        else:
+            return self.simpleEvaluateAtWavelength(wave)
+
+    def _imageAtWavelength(self, wave):
+        """
+        Get an image of the object at a particular wavelength, using linear interpolation between
+        the originally-stored images.
+        """
+        # First, some wavelength-related sanity checks.
+        if self.waves is None:
+            raise RuntimeError("Requested image at some wavelength when doing direct calculation!")
+        if wave < min(self.waves) or wave > max(self.waves):
+            raise RuntimeError("Requested wavelength %.1f is outside the allowed range:"
+                               " %.1f to %.1f nm"%(wave, min(self.waves), max(self.waves)))
+
+        # Figure out where the supplied wavelength is compared to the list of wavelengths on which
+        # images were originally tabulated.
+        lower_idx = np.searchsorted(self.waves, wave)-1
+        # There can be edge issues, so watch out for that:
+        if lower_idx < 0: lower_idx = 0
+        if lower_idx == len(self.waves)-1: lower_idx = len(self.waves)-1
```

Comment:
Oh my.  Thanks for the careful reading. :)  You are right, of course.  Fix pushed.